hstream-server/src/HStream/Server/ClickHouseConnector.hs

```diff
@@ -0,0 +1,124 @@
+import qualified RIO.Map                          as M
+import qualified Z.Data.CBytes                    as ZCB
+import qualified Z.Data.JSON                      as JSON
+
+import           HStream.SQL.Codegen
+import           HStream.Processing.Util 
+import           System.IO.Unsafe
+import           System.IO
+
+import Database.ClickHouseDriver
+import Database.ClickHouseDriver.Types
+import Haxl.Core (Env(states))
+import qualified Data.HashMap.Strict                             as HM
+import qualified Data.Map.Strict as DMS
+import           Data.Aeson
+import HStream.SQL.AST
+import Data.Aeson.Types
+import Data.Scientific
+import qualified Data.ByteString.Char8 as DBC
+
+
+main :: IO ()
+main = do
+  handleInsertSQL $ "INSERT INTO " <> "source1" <> " (temperature, humidity) VALUES (22, \"80\");"
+
+defaultCKClient :: IO(Env () w)
+defaultCKClient = createClient ConnParams{
+       username'    = "default"
+      ,host'        = "host.docker.internal"
+      ,port'        = "9000"
+      ,password'    = ""
+      ,compression' = False
+      ,database'    = "default"
+    }
+
+handleInsertSQL :: Text -> IO ()
+handleInsertSQL sql = do
+  plan <- streamCodegen sql
+  case plan of
+    InsertPlan topicName payload -> do
+      timestamp <- getCurrentTimestamp
+      writeRecord
+        (clickHouseSinkConnector defaultCKClient)
+        SinkRecord
+          { snkStream = topicName,
+            snkKey = Nothing,
+            snkValue = payload,
+            snkTimestamp = timestamp
+          }
+    _ -> error "Execution plan type mismatched"
+
+clickHouseSinkConnector :: IO(Env () w) -> SinkConnector
+clickHouseSinkConnector ckClient = SinkConnector {
+  writeRecord = writeRecordToClickHouse ckClient
+}
+
+valueToCKType :: Value -> ClickhouseType  
+valueToCKType (String s) = CKString (DBC.pack $ show s)
+valueToCKType (Bool b) = if b then CKInt8 1 else CKInt8 0
+valueToCKType (Number sci) = do
+  case floatingOrInteger sci of
+    Left r -> CKDecimal64 r 
+    Right i -> CKInt64 i
+
+writeRecordToClickHouse :: IO(Env () w) -> SinkRecord -> IO ()
+writeRecordToClickHouse ckClient SinkRecord{..} = do
+    conn <- ckClient
+    ping conn
+    print snkValue
+    let insertMap = (decode snkValue) :: Maybe (Map String Value)
+    -- print $ show tmp
+    case insertMap of 
+      -- Just l -> print $ toList l
+      Just l -> do
+        print $ DMS.keys l
+        let keys = DMS.keys l
+        let elems = DMS.elems l
+        insertOneRow conn ("INSERT INTO " ++ show snkStream ++ " " ++ keysAsSql keys ++" VALUES ") (valueToCKType <$> elems)
+      -- _ -> return ()
+
+    -- logId <- getCLogIDByStreamName ckClient (textToCBytes snkStream)
+    -- let payload =
+    --       Payload {
+    --         pTimestamp = snkTimestamp,
+    --         pKey = fmap lazyByteStringToCbytes snkKey,
+    --         pValue = lazyByteStringToCbytes snkValue
+    --       }
+    -- lsn <- appendSync ckClient logId (JSON.encode payload) Nothing
```

Comment:
What does this do?

Reply:
W.I.P 
But I'm not sure about what should I do with logId? Do we need it in clickhouse?